custom_components/localtuya/light.py

```diff
@@ -187,24 +202,26 @@ def __init__(
         self._effect_list = []
         self._scenes = {}
 
-        custom_scenes = False
         if self.has_config(CONF_SCENE):
-            if self.has_config(CONF_SCENE_VALUES):
-                custom_scenes = True
+            if self.has_config(CONF_SCENE_VALUES) and len(self._config.get(CONF_SCENE_VALUES)):
                 values_list = list(self._config.get(CONF_SCENE_VALUES))
                 values_name = list(self._config.get(CONF_SCENE_VALUES).values())
                 self._scenes = dict(zip(values_name, values_list))
+            elif self._write_only: # BLE bulbs
+                self._scenes = SCENE_LIST_RGBW_BLE
             elif int(self._config.get(CONF_SCENE)) < 20:
                 self._scenes = SCENE_LIST_RGBW_255
```

Comment:
@xZetsubou I'd fix this broken logic as well, but I don't know how to do it 😞 

`SCENE_LIST_RGBW_255` has data format [documented](https://developer.tuya.com/en/docs/iot/dj?id=K9i5ql3v98hn3#title-10-scene_data) for a DP named `scene_data`. `SCENE_LIST_RGB_1000` and `SCENE_LIST_RGB_1000` are for DP named `scene_data_v2`. I don't know how to tell here from which name the DP was found. Please help if you can!